das/options.go

```diff
@@ -0,0 +1,114 @@
+package das
+
+import (
+	"fmt"
+	"time"
+)
+
+var (
+	ErrInvalidOption = fmt.Errorf("das: invalid option")
+)
+
+func errInvalidOptionValue(optionName string, value string) error {
+	return fmt.Errorf("%w: value %s cannot be %s", ErrInvalidOption, optionName, value)
+}
+
+// Type Option is the functional option that is applied to the daser instance for parameters configuration.
+type Option func(*DASer) error
+
+// Type parameters is the set of parameters that must be configured for the daser
+type parameters struct {
+	samplingRange     uint64
+	concurrencyLimit  int
+	bgStoreInterval   time.Duration
+	priorityQueueSize int
+	genesisHeight     uint64
+}
+
+// defaultParameters returns the default configuration values for the daser parameters
+func defaultParameters() parameters {
+	return parameters{
+		samplingRange:     100,
+		concurrencyLimit:  16,
+		bgStoreInterval:   10 * time.Minute,
+		priorityQueueSize: 16 * 4,
+		genesisHeight:     1,
+	}
+}
+
+// WithSamplingRange is a functional option to configure the daser's `samplingRange` parameter
+// ```
+//
+//	WithSamplingRange(10)(daser)
+//
+// ```
+// or
+// ```
+//
+//	option := WithSamplingRange(10)
+//	// shenanigans to create daser
+//	option(daser)
+//
+// ```
+func WithSamplingRange(samplingRange uint64) Option {
+	return func(d *DASer) error {
+		if samplingRange == 0 {
+			return errInvalidOptionValue(
+				"samplingRange",
+				"0",
+			)
+		}
+
+		d.params.samplingRange = samplingRange
+
+		return nil
+	}
+}
+
+// WithConcurrencyLimit is a functional option to configure the daser's `concurrencyLimit` parameter
+// Refer to WithSamplingRange documentation to see an example of how to use this
+func WithConcurrencyLimit(concurrencyLimit int) Option {
+	return func(d *DASer) error {
+		if concurrencyLimit == 0 {
```

Comment:
This check does not check "-" values, while `int` allows them. Also true for other options.